app/src/main/java/com/github/swent/echo/compose/event/EventTagEntry.kt

```diff
@@ -1,42 +1,72 @@
 import androidx.compose.foundation.rememberScrollState
 import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
 import androidx.compose.material.icons.filled.Clear
 import androidx.compose.material3.Icon
+import androidx.compose.material3.OutlinedIconButton
 import androidx.compose.material3.Text
 import androidx.compose.material3.TextButton
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.stringResource
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.github.swent.echo.R
+import com.github.swent.echo.compose.components.TagSelectionDialog
 import com.github.swent.echo.data.model.Tag
 
 /** this composable contains the Tags title, text field and list of tags */
 @Composable
 fun EventTagEntry(
     tags: Set<Tag>,
-    tagText: String,
-    onTagFieldChanged: (tagText: String) -> Unit,
-    onTagPressed: (tag: Tag) -> Unit
+    onTagSelected: (tag: Tag) -> Unit,
+    onTagDeleted: (tag: Tag) -> Unit
 ) {
-    EventTextEntry(
-        name = stringResource(R.string.edit_event_screen_tags),
-        value = tagText,
-        onValueChange = onTagFieldChanged
-    )
-    Row(modifier = Modifier.horizontalScroll(rememberScrollState())) {
-        tags.forEach { tag ->
-            TextButton(onClick = { onTagPressed(tag) }) {
-                Text(tag.name)
-                Icon(
-                    imageVector = Icons.Filled.Clear,
-                    contentDescription =
-                        stringResource(R.string.edit_event_screen_delete_tag_button) + tag.name
-                )
+    var dialogVisible by remember { mutableStateOf(false) }
+    Column(modifier = Modifier.fillMaxWidth().padding(EVENT_PADDING_BETWEEN_INPUTS)) {
+        EventEntryName(stringResource(R.string.edit_event_screen_tags))
+        Row(modifier = Modifier.horizontalScroll(rememberScrollState())) {
+            tags.forEach { tag ->
+                TextButton(
+                    onClick = { onTagDeleted(tag) },
+                    modifier = Modifier.testTag("${tag.name}-tag-button")
+                ) {
+                    Text(tag.name)
+                    Icon(
+                        imageVector = Icons.Filled.Clear,
+                        contentDescription =
+                            stringResource(R.string.edit_event_screen_delete_tag_button) + tag.name
+                    )
+                }
             }
+            OutlinedIconButton(
+                modifier = Modifier.testTag("add-tag-button"),
+                onClick = { dialogVisible = true }
+            ) {
+                val icon = Icons.Filled.Add
+                Icon(imageVector = icon, contentDescription = icon.name)
+            }
```

Comment:
the button to add a new tag should also have a label next to the plus icon; at least "Add" or better "Add Tag" to make the button accessible for screen reader users:
![image](https://github.com/echo-swent/echo/assets/22591354/63b113d5-ca53-4bea-a4c3-7bd93180424c)

Reply:
That's the role of the `contentDescription`, but it's true that the icon description isn't very descriptive